core/src/main/java/io/questdb/griffin/SqlUtil.java

```diff
@@ -241,6 +241,43 @@ public static CharSequence fetchNext(GenericLexer lexer) throws SqlException {
         return null;
     }
 
+    public static ExpressionNode findSampleByNode(QueryModel model) {
+        while (model != null) {
+
+            if (SqlUtil.isNotWrapperModel(model)) {
+                return null;
+            }
+
+            final ExpressionNode sampleBy = model.getSampleBy();
+            if (sampleBy != null && sampleBy.type == ExpressionNode.CONSTANT) {
+                return sampleBy;
+            }
+
+            model = model.getNestedModel();
+        }
+        return null;
+    }
+
+    public static QueryColumn findTimestampFloorColumn(QueryModel model) {
```

Comment:
This method seems to be doing more than just finding timestamp_floor column as it also descends through models. This deserves to be highlighted in the method name or in javadoc. The same applies to `findSampleByNode`. Alternatively, we could keep these two methods in `CreateMatViewOperationImpl`. WDYT?